darts/models/forecasting_model.py

```diff
@@ -644,3 +644,72 @@ def predict(self,
         if self._expect_covariates and covariates is None:
             raise_log(ValueError('The model has been trained with covariates. Some matching covariates '
                                  'have to be provided to `predict()`.'))
+
+
+class ExtendedForecastingModel(ForecastingModel, ABC):
+    """ The base class for "extended" forecasting models, handling optional covariates.
+
+    All implementations have to implement the `fit()` and `predict()` methods defined below.
+    The `fit()` method is meant to train the model on a time series, along with optional
+    covariates.
+    """
+
+    _expect_covariates = False
+
+    @abstractmethod
+    def fit(self,
+            series: TimeSeries,
+            covariates: Optional[Union[TimeSeries]] = None
+            ) -> None:
+        """ Fits/trains the model on the provided series
+
+        Defines behavior that should happen when calling the `fit()` method of every forecasting model.
+
+        The models can handle covariates.
+
+        Parameters
+        ----------
+        series
+            A time series. The model will be trained to forecast this time series.
+        covariates
+            A time series of covariates. This time series will not be forecasted, but can be used by
+            some models as an input. Some of the covariates may represent forecasts known in advance. This knowledge
```

Comment:
Could you remove the part saying `Some of the covariates may represent forecasts known in advance. This knowledge is a property of the `TimeSeries`.` ?
It's something I wrote during development and forgot to remove, but it's not quite accurate.

Reply:
Done. I have also removed it from `GlobalForecastingModel`.